### ISSUE TITLE
Other non-application tests

### ISSUE THREAD

**Issue by Fryguy** [ManageIQ/manageiq-debug#1]
I've been thinking it might be helpful to have some diagnostics for basic environment stuff, like IO, kind of like https://www.thomas-krenn.com/en/wiki/Linux_I/O_Performance_Tests_using_dd

@carbonin WDYT?

**Comment by carbonin** [comment]
Yeah, if we can find something that we know maps to the performance of our application or that we know can vary depending on kube installation methods then I'd be for it.

**Comment by carbonin** [comment]
I think it would make sense to bake a big-ish file into the image somewhere then do a read test (`dd if=<$GEM_DIR/big/ish/file> of=/dev/null` or some such) against that.
That way it's guaranteed to not be in some cached layer (as something that we just wrote out might be) and it's as close to the behavior of what we think is the problem (`rake environment`) as can be.

I'm not sure write performance is quite as important, but we could include that too for completeness.

**Comment by bdunne** [comment]
I don't think local filesystem I/O performance is important.  Once the code is loaded, it shouldn't make much of a difference.  If we want to test storage performance for something like postgres, we would have to grab a PV.  If we grab any random PV (the default) there's no guarantee that it's using the same backend storage as the postgres PVC.  If we grab the postgres PVC, then we have to take postgres and the application down in order to use the claim since it's `ReadWriteOnce`.

**Comment by Fryguy** [comment]
yeah, the thought is that it's really the PV performance (i.e. postgres).  Even so, in some environments we are seeing even code load being expensive.

**Comment by bdunne** [comment]
I think the commands that we currently have are a good representation of code loading performance.  We may want to add timing running the liveness probe.  I found that to be helpful a couple days ago.

**Comment by bdunne** [comment]
Maybe we can add the I/O performance testing tools to the postgres container.  Then we can `rsh` in and run it.

**Comment by Fryguy** [comment]
This is interesting... https://www.postgresql.org/docs/devel/pgbench.html

Wonder if we should add that to debug and/or just replace our db ping command with that.
